fe2-android/app/src/main/java/com/github/dedis/student20_pop/model/Election.java

```diff
@@ -78,12 +128,39 @@ public long getStartTimestamp() {
     }
 
     @Override
-    public EventType getType() {
-        return EventType.ELECTION;
+    public long getEndTimestamp() {
+        return end;
+    }
+
+    /**
+     * Computes the hash for the registered votes, when terminating an election (sorted by message id's alphabetical order)
+     * @return the hash of all registered votes
+     */
+    public String computerRegisteredVotes() {
+        List<String> listOfVoteIds = new ArrayList<>();
+        //Since messageMap is a TreeMap, votes will already be sorted in the alphabetical order of messageIds
+        for (String senderPk: messageMap.values()) {
+            for (ElectionVote vote: voteMap.get(senderPk)) {
+                listOfVoteIds.add(vote.getId());
+            }
+        }
+        if (listOfVoteIds.isEmpty()) return Hash.hash("novotes");
```

Comment:
Is this convention of hashing "novotes" decided for when there are no votes?

Reply:
I put this just to prevent an election without any votes to make the hashing process crash, but after further discussion with the other teams, I'll just send an empty string